src/rulesets/oas2/__tests__/valid-example.ts

```diff
@@ -156,7 +156,7 @@ describe('valid-example', () => {
       expect.objectContaining({
         code: 'valid-example',
         message: '"self" property type should be array',
-        path: ['definitions', 'halRoot', '_links', 'self'],
+        path: ['definitions', 'halRoot', 'example', '_links', 'self'],
```

Comment:
This is quite tricky. I still need to fix it.

Reply:
This is a bit of a different issue, fixed in https://github.com/stoplightio/spectral/pull/459/commits/0e3844f2025cd5fb07d41f217e1c33c5e3d85ef0
yet I believe the new error message we generate is actually worse than the former one, or am I wrong?